tests/unit_test.py

```diff
@@ -0,0 +1,142 @@
+from enums import Player
+from unittest.mock import patch, Mock
+from Piece import Knight
+from Piece import Pawn
+
+
+# Test functions to create mock game state for testing
+def empty_board(row: int, col: int) -> Player:
+    """
+    Returns empty board
+    """
+    return Player.EMPTY
+
+
+def piece_in_all_possibilities(row: int, col: int) -> Player:
+    """
+    Returns board with a pawn at Player.PLAYER_2
+    """
+    return Pawn("my_pawn", row, col, Player.PLAYER_2)
+
+
+def piece_in_most_possibilities(row: int, col: int) -> Player:
+    """
+    Returns board with a pawn at Player.PLAYER_2, except at (1, 2)
+    """
+    return Player.EMPTY if (row, col) == (1, 2) else Pawn("my_pawn", row, col, Player.PLAYER_2)
+
+
+def piece_of_player2(row: int, col: int) -> Player:
+    """
+    Return a Pawn belonging to Player.PLAYER_1 at (1,2) and a Pawn belonging to Player.PLAYER_2 at all other positions
+    """
+    return Pawn("my_pawn", row, col, Player.PLAYER_1) if (row, col) == (1, 2) else Pawn("my_pawn", row, col,
+                                                                                        Player.PLAYER_2)
+
+
+def there_is_not_a_piece(row: int, col: int) -> bool:
+    """
+    Return False to indicate that there is no piece at the given position
+    """
+    return False
+
+
+def there_is_a_piece(row: int, col: int) -> bool:
+    """
+    Return True to indicate that there is a piece at the given position
+    """
+    return True
+
+
+def test_get_valid_peaceful_moves():
+    """
```

Comment:
I would use documentation instead, comments make it look messy
https://swimm.io/blog/code-comments-vs-documentation/

Reply:
Got fixed